datalib/prepare_webdataset_local.py

```diff
@@ -0,0 +1,61 @@
+from tqdm import tqdm
+from glob import glob
+from pathlib import Path
+import webdataset as wds
+from datasets import load_dataset
+from huggingface_hub import upload_file
+import os
+import io
+
+from datalib.mappings_local import prepare_local_dataset
+
+def generate_wds_samples(dsname,channel_name,language):
+    samples = []
+    folder_path="/mnt/disks/audio-data-2/Data/"
+    channel_path=os.path.join(folder_path,channel_name)
+    audio_path=os.path.join(channel_path,"audio_files_compressed/")
+    for file in os.listdir(audio_path):
+        filename = os.path.basename(file).replace(".wav","")
+        sample = prepare_local_dataset(dsname=dsname,channel_name=channel_name,audio_name=filename,folder_path=folder_path,split='train',language=language)
+        samples.append(sample)
```

Comment:
this is a generator, you should yield here instead of adding to a list and then return